File: lecture notes/matlab/lec3/run1.m
clear;clc;
%%%%%%%%%%%%%%%%%%%%%%%%
m=1;
b=1;
k=1;

Gs=tf(1,[m b k]);
[y,t]=step(Gs,10);
%%%%%%%%%%%%%%%%%%%%%%%%
T=0.1;
fac1=(2*m+b*T)/(m+b*T+k*T^2);
fac2=-m/(m+b*T+k*T^2);
fac3=T^2/(m+b*T+k*T^2);

tvec=0:T:10;
xt=zeros(size(tvec));
ut=ones(size(tvec));
for i=3:length(tvec)
    xt(i)=fac1*xt(i-1)+fac2*xt(i-2)+fac3*ut(i);
end
%%%%%%%%%%%%%%%%%%%%%%%%
T=0.5;
fac1=(2*m+b*T)/(m+b*T+k*T^2);
fac2=-m/(m+b*T+k*T^2);
fac3=T^2/(m+b*T+k*T^2);

tvec2=0:T:10;
xt2=zeros(size(tvec2));
ut2=ones(size(tvec2));
for i=3:length(tvec2)
    xt2(i)=fac1*xt2(i-1)+fac2*xt2(i-2)+fac3*ut2(i);
end
%%%%%%%%%%%%%%%%%%%%%%%%
figure(1);clf;hold on;grid minor;ax1=gca;
xlabel("Zaman(s)");ylabel("y(t)");title("Yay-kütle-damper yanıtı");
set(gca,'MinorGridColor','k','MinorGridAlpha',1);
legend("show");
plot(t,y,'k','LineWidth',2,'DisplayName','step');
stem(tvec,xt,'r','LineWidth',2,'DisplayName',"fark denklemi(T=0.1)",'Marker','none');
stem(tvec2,xt2,'b','LineWidth',2,'DisplayName',"fark denklemi(T=0.5)",'Marker','none');
print("../../img/"+"lec3_step1.eps",'-depsc','-r150');
%%%%%%%%%%%%%%%%%%%%%%%%
T=0.25;
Gz=c2d(tf(1,[m b k]),T);
[yz,tz]=step(Gz,10);

T=1;
Gz2=c2d(tf(1,[m b k]),T);
[yz2,tz2]=step(Gz2,10);

figure(2);clf;hold on;grid minor;ax2=gca;
set(gca,'MinorGridColor','k','MinorGridAlpha',1);
legend("show");
xlabel("Zaman(s)");ylabel("y(t)");title("Yay-kütle-damper yanıtı");
plot(t,y,'k','LineWidth',2,'DisplayName','step');
stem(tz,yz,'r','LineWidth',2,'DisplayName',"fark denklemi(T=0.25)",'Marker','none');
stem(tz2,yz2,'b','LineWidth',2,'DisplayName',"fark denklemi(T=1)",'Marker','none');
print("../../img/"+"lec3_step2.eps",'-depsc','-r150');
%%%%%%%%%%%%%%%%%%%%%%%%

File: lecture notes/python/lec3/run1.py
import matplotlib.pyplot as plt
import numpy as np
import sympy as sym
import tbcontrol
import control
############################
# q1
############################
def fun_q1():
    m=1
    b=1
    k=1
    T=0.1
    fac1=(2*m+b*T)/(m+b*T+k*T**2)
    fac2=-m/(m+b*T+k*T**2)
    fac3=T**2/(m+b*T+k*T**2)
    tvec=np.arange(0,10+1,T)
    xt=np.zeros(tvec.shape)
    ut=np.ones(tvec.shape)
    for i in range(0,len(tvec)):
        if i==0:
            xt[i]=fac1*0+fac2*0+fac3*0
        elif i==1:
            xt[i]=fac1*xt[i-1]+fac2*0+fac3*ut[i-1]
        else:
            xt[i]=fac1*xt[i-1]+fac2*xt[i-2]+fac3*ut[i-1]

    plt.grid('minor')
    plt.xlabel("Zaman(s)")
    plt.ylabel("x(t)")
    plt.title("Yay-kütle-damper sistem yanıtı")

    Gz=control.tf(1,np.array([m,b,k]))
    tc, yc=control.step_response(Gz)
    plt.plot(tc,yc,'k')
    plt.stem(tvec,xt,'b')
    plt.show()
############################
# q2
############################
def fun_q2():
    m=1
    b=1
    k=1
    
    plt.grid('minor')
    plt.xlabel("Zaman(s)")
    plt.ylabel("x(t)")
    plt.title("Yay-kütle-damper sistem yanıtı")

    Gz=control.tf(1,np.array([m,b,k]))
    tc, yc=control.step_response(Gz)

    Gz1=control.c2d(control.tf(1,np.array([m,b,k])),0.1)
    tc1, yc1=control.step_response(Gz1)

    Gz2=control.c2d(control.tf(1,np.array([m,b,k])),0.5)
    tc2, yc2=control.step_response(Gz2)

    plt.plot(tc,yc,'k')
    plt.stem(tc1,yc1,'r')
    plt.stem(tc2,yc2,'b')
    plt.show()
fun_q2()